packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -21,11 +22,18 @@ export class CheckboxBase extends React.Component<ICheckboxProps, ICheckboxState
   private _classNames: { [key in keyof ICheckboxStyles]: string };
 
   public static getDerivedStateFromProps(props: ICheckboxProps, state: ICheckboxState): ICheckboxState {
-    if (props.checked !== undefined) {
+    if (props.indeterminate !== undefined && state.isIndeterminate) {
       return {
         ...state,
-        isChecked: !!props.checked
+        isIndeterminate: !!props.indeterminate
       };
+    } else {
```

Comment:
because you are returning in the first if statement you can drop the `else` and just do another if statement right after.